### PR TITLE
Move `new` operator and some class setup boilerplate to lualib

### LINKED ISSUE #709: Extract new operator and class creation boilerplate to a lualib
Was discussed in Discord a while back.

Input:
```ts
class A {}
new A(args);
```

Current result:

```lua
A = ...
function A.new(...)
    local self = setmetatable({}, A.prototype)
    self:____constructor(...)
    return self
end
A.new(args)
```

Proposed Result:

```lua
function __TS_New(class, ...)
    local self = setmetatable({}, class.prototype)
    return self:____constructor(...) or self
end

A = ...
__TS_New(A, args)
```

---

Input:

```ts
class A {}
class B extends A {}
```

Current Result:

```lua
A = {}
A.name = "A"
A.__index = A
A.prototype = {}
A.prototype.__index = A.prototype
A.prototype.constructor = A
B = {}
B.name = "B"
B.__index = B
B.prototype = {}
B.prototype.__index = B.prototype
B.prototype.constructor = B
B.____super = A
setmetatable(B, B.____super)
setmetatable(B.prototype, B.____super.prototype)
```

Proposed Result:

```lua
function __TS_CreateClass()
    local class = {}
    class.__index = class
    class.name = name
    class.prototype = {}
    class.prototype.__index = class.prototype
    class.prototype.constructor = class
    return class
end

function __TS_ClassExtends(class, base)
	class.____super = base
    setmetatable(class, class.____super)
    setmetatable(class.prototype, class.____super.prototype)
end

A = __TS_CreateClass()
A.name = "A"
B = __TS_CreateClass()
B.name = "B"
__TS_ClassExtends(B, A)
```

### PR DESCRIPTION
**Pull request by ark120202**
Resolves #709.

The changes are relatively small, because of how much things depend on accessors. I want to change that later, but that'd be out of the scope of this PR.

### CODE REVIEW

File: test/translation/__snapshots__/transformation.spec.ts.snap [hunk 1]
```diff
@@ -40,17 +40,10 @@ end"
 `;
 
 exports[`Transformation (classPureAbstract) 1`] = `
-"ClassB = {}
+"require(\\"lualib_bundle\\");
+ClassB = __TS__Class()
 ClassB.name = \\"ClassB\\"
-ClassB.__index = ClassB
-ClassB.prototype = {}
 ClassB.prototype.__index = ClassB.prototype
```
Comment (Perryvw) [inline test/translation/__snapshots__/transformation.spec.ts.snap]: What is the reason this line didn't move to __TS__Class?

File: test/unit/printer/sourcemaps.spec.ts [hunk 2]
```diff
@@ -84,16 +84,12 @@ test.each([
         `,
 
         assertPatterns: [
-            { luaPattern: "Bar = {}", typeScriptPattern: "class Bar" },
+            { luaPattern: "Bar =", typeScriptPattern: "class Bar" },
```
Comment (Perryvw) [inline test/unit/printer/sourcemaps.spec.ts]: could you make the luaPattern `Bar = __TS__Class()`?

File: test/unit/decorators/tupleReturn.spec.ts [hunk 3]
```diff
@@ -404,107 +404,7 @@ test("TupleReturn method assignment", () => {
     `;
 
     const lua = util.transpileString(code);
-    expect(lua).toBe("local jkl = def.new()\nlocal a, b = jkl:abc()");
-});
-
-test("TupleReturn functional", () => {
```
Comment (Perryvw) [inline test/unit/decorators/tupleReturn.spec.ts]: Why were these tests removed?

Comment (ark120202) [reply test/unit/decorators/tupleReturn.spec.ts]: They were duplicated, I think I accidentally pasted them twice during some refactor

File: test/translation/__snapshots__/transformation.spec.ts.snap (hunk 1 above)
Comment (ark120202) [reply test/translation/__snapshots__/transformation.spec.ts.snap]: It's added conditionally, only when there are no getters. Though probably it'd be better to move it as well, an extra assignment wouldn't be a problem anyway, given how slow getters already are

**Pull request merged**